include/miral/miral/window_manager_tools.h

```diff
@@ -39,229 +39,273 @@ struct ApplicationInfo;
- *  @{ */
-
-    /** count the applications
-     *
-     * @return number of applications
-     */
+    /// Returns the number of applications.
+    ///
+    /// @returns the number of applications
     auto count_applications() const -> unsigned int;
 
-    /** execute functor for each application
-     *
-     * @param functor the functor
-     */
+    /// Execute a functor for each application.
+    ///
+    /// \param functor called for each application
     void for_each_application(std::function<void(ApplicationInfo& info)> const& functor);
 
     /** find an application meeting the predicate
      *
      * @param predicate the predicate
      * @return          the application
      */
+    /// Find an application that meets the predicate.
+    ///
+    /// \param predicate the predicate
+    /// \returns an application, or the default-constructed application if none is found
     auto find_application(std::function<bool(ApplicationInfo const& info)> const& predicate)
     -> Application;
 
-    /** retrieve metadata for an application
-     *
-     * @param session   the application session
-     * @return          the metadata
-     */
+    /// Retrieve information about a \p session.
+    ///
+    /// \param session the application
+    /// \returns info about the application, or the default-constructed application info if the \p session is invalid
+    /// \sa miral::ApplicationInfo - info about the application
     auto info_for(std::weak_ptr<mir::scene::Session> const& session) const -> ApplicationInfo&;
 
-    /** retrieve metadata for a window
-     *
-     * @param surface   the window surface
-     * @return          the metadata
-     */
+    /// Retrieve info about a window from the \p surface.
+    ///
+    /// \param surface the surface
+    /// \returns info about the window, or the default-constructed window info if the \p surface is invalid
+    /// \sa miral::WindowInfo - info about the window
     auto info_for(std::weak_ptr<mir::scene::Surface> const& surface) const -> WindowInfo&;
 
-    /** retrieve metadata for a window
-     *
-     * @param window    the window
-     * @return          the metadata
-     */
+    /// Retrieve info about a window.
+    ///
+    /// \param window the window
+    /// \returns info about the window, or the default-constructed window info if the \p surface is invalid
+    /// \throws std::out_of_range when the window is invalid
+    /// \sa miral::WindowInfo - info about the window
     auto info_for(Window const& window) const -> WindowInfo&;
 
-    /** retrieve metadata for a persistent surface id
-     *
-     * @param id        the persistent surface id
-     * @return          the metadata
-     * @throw           invalid_argument or runtime_error if the id is badly formatted/doesn't identify a current window
-     * \deprecated      'Persistent' surface IDs were part of mirclient API
-     */
+    /// Retrieve metadata for a persistent surface id.
```

Comment:
Elsewhere "metadata" has been rephrased to "info" is there a reason for this inconsistency?